python-frontend/src/test/java/org/sonar/python/semantic/ClassSymbolTest.java

```diff
@@ -315,6 +318,35 @@ public void copy_without_usages() {
     assertEqualsWithoutUsages(classSymbol);
   }
 
+  @Test
+  public void static_member_usages() {
+    ClassSymbolImpl classSymbol = ((ClassSymbolImpl) lastClassSymbol(
+      "class A:",
+      "  foo = 42",
+      "  def __init__(self): ",
+      "    A.foo",
+      "    A.bar"
+    ));
+    Map<String, Symbol> members = classSymbol.declaredMembers().stream().collect(Collectors.toMap(Symbol::name, Function.identity()));
+    Symbol foo = members.get("foo");
+    assertThat(foo.usages()).extracting(Usage::kind).containsExactly(Usage.Kind.ASSIGNMENT_LHS, Usage.Kind.OTHER);
+
+    assertThat(members.get("bar")).isNull();
+
+    // receiver is not a class symbol
+    classSymbol = ((ClassSymbolImpl) lastClassSymbol(
+      "class A:",
+      "  foo1 = 42",
+      "  def __init__(self, x): ",
+      "    getClass().foo2", // receiver hasn't a symbol
+      "    x.foo3",          // receiver symbol isn't a class
+      "    unknown.foo4"     // receiver symbol unknown
+    ));
+    assertThat(classSymbol.declaredMembers())
+      .filteredOn("kind", Symbol.Kind.OTHER)
+      .extracting(Symbol::name).containsOnly("foo1");
```

Comment:
The intention of this assertion is not quite clear to me.

Reply:
The intention is asserting that the only static member of `A` is `foo1` and that it doesn't contain `foo2`, `foo3` and `foo4`.
I will try to make that more explicit